### PR TITLE
Feature 39 As a user I want to have an ability to see all my folders/files

### PR DESCRIPTION
**Pull request by IvanitskyiVolodymyr**
Additionally, where necessary, I added an avatar from the database, and if it is null, the default avatar is displayed.

### CODE REVIEW

**Comment by IvanitskyiVolodymyr** [comment]
image

**Comment by IvanitskyiVolodymyr** [comment]
image

File: backend/Recodo.BLL/Services/FolderService.cs [hunk 1]
```diff
@@ -23,5 +26,50 @@ public async Task<FolderDTO> CreateFolder(NewFolderDTO newFolderDTO)
...
+                ToListAsync();
+
+            if(foldersEntities.Count == 0)
+            {
+                throw new NotFoundException(nameof(Folder));
+            }
+
+            var folders = _mapper.Map<List<FolderDTO>>(foldersEntities);
+
+            return folders;
```
Comment (Anatolii-Ivanov) [inline backend/Recodo.BLL/Services/FolderService.cs]: Don't create variable if not needed
return _mapper..

File: backend/Recodo.BLL/Services/FolderService.cs [hunk 2]
```diff
@@ -23,5 +26,50 @@ public async Task<FolderDTO> CreateFolder(NewFolderDTO newFolderDTO)
...
+        }
+
+        public async Task Delete(int id)
+        {
+            if (id < 0)
+                throw new NotFoundException(nameof(Folder), id);
```
Comment (Anatolii-Ivanov) [inline backend/Recodo.BLL/Services/FolderService.cs]: Use the same formatting 
Or
if ()
{
    something
{

Or
if ()
    something

File: frontend/src/app/models/user/user-dto.ts [hunk 3]
```diff
@@ -2,4 +2,5 @@ export interface UserDto {
     id : number;
     email : string;
     workspaceName : string;
+    avatarLink : string;
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/models/user/user-dto.ts:5]: Also find one formatting time, somewhere it is
avatarLink:string
Somewhere
avatarLink : string

File: frontend/src/app/modules/workspace/personal/personal.component.ts [hunk 4]
```diff
@@ -44,58 +22,153 @@ const ELEMENT_DATA = [
...
 
   ngOnInit(): void {
     this.getAutorithedUser();
   }
 
+  private getFolders() {
+    return this.folderService.getAllFoldersByUserId(this.currentUser.id).subscribe(
+      (result) => {
+        this.folders = result;
+        var name = this.folders.find(f => f.id == this.selectedFolderId)?.name;
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/workspace/personal/personal.component.ts]: No vars

File: frontend/src/app/modules/workspace/personal/personal.component.ts [hunk 5]
```diff
@@ -44,58 +22,153 @@ const ELEMENT_DATA = [
...
+  }
 
-    dialogRef.afterClosed().subscribe((result) => {
-      this.createFolder(result);
-    });
+  showNewFolderForm(folder?: FolderDto) {
+    const dialogConfig = new MatDialogConfig();
+
+    dialogConfig.autoFocus = true;
+    dialogConfig.data = folder === undefined ? '' : folder.name;;
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/workspace/personal/personal.component.ts]: const dialogConfig : MatDialogConfig = {
autoFocus = true,
data = ...
}

Should be something like this

File: frontend/src/app/services/folder.service.ts [hunk 6]
```diff
@@ -21,10 +21,26 @@ export class FolderService extends ResourceService<NewFolderDto> {
...
+  public deleteFolder(id: number) {
+    return this.delete(id);
+  }
+
+  public getAllFoldersByUserId(id: number) {
+    return this.getFullRequest<NewFolderDto>(`folders/${id}`).pipe(
+      map((resp) => {
+        return resp.body as unknown as FolderDto[];
+      })
+    );
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/services/folder.service.ts]: I think getFullRequest<FolderDto[]>

File: frontend/src/app/services/time.service.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TimeService {
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/services/time.service.ts:6]: You can use moment library, but ok

File: backend/Recodo.BLL/Services/FolderService.cs (hunk 1 above)
Comment (IvanitskyiVolodymyr) [reply backend/Recodo.BLL/Services/FolderService.cs]: Agree, corrected

File: backend/Recodo.BLL/Services/FolderService.cs (hunk 2 above)
Comment (IvanitskyiVolodymyr) [reply backend/Recodo.BLL/Services/FolderService.cs]: Fixed, now looks more beautiful

**Review by Anatolii-Ivanov — APPROVED** [review]

**Pull request merged**